README.md

```diff
@@ -1,2 +1,331 @@
+
+## 4.2 Casts
+
+A Cast is a public message created by a user that is displayed on their profile.  It can contain text as well as links to media, on-chain activity or other casts. Users are allowed to delete casts at any time. 
+
+Casts are managed using a two phase set, which has two collections: an **add set** and a **remove set**. New casts are placed in the add set and moved to the remove set when deleted by the user. Once a message is deleted, it can never be moved back into the add set. The Cast Set can be said to have "remove wins" semantics for the merge operation. 
+
+Casts come in several different flavors and the protocol can be extended to support many more types in the future. Each type is stored in it's own two-phase set which may have additional rules for merging new messages into the set. 
+
+### 4.2.1 Short Text Casts
+A short public post created by a user that can appear directly on their profile, as a reply to another cast, url or on-chain item. Short casts can have upto 280 unicode characters and two embeds. The `parentUri` property can reference any URI, except for itself or its children. 
+
+```ts
+type CastShortTextBody = {
+  embed: Embed;
+  text: string;
+  schema: 'farcaster.xyz/schemas/v1/cast-short-text';
+  parentUri?: URI;
+};
+```
+
+### 4.2.2 Recasts
+A recast expresses an intent to share another cast. The set ensures that a recast message does not reference itself, and that only one recast exists for each value of `account` and `targetCastUri`. If multiple values are discovered, it keeps the one with the highest timestamp and highest lexicographical order, in that order. 
+
+```ts
+type CastRecastMessageBody = {
+  targetCastUri: URI;
+  schema: 'farcaster.xyz/schemas/v1/cast-recast';
+};
+```
+
+### 4.2.3 Deletes
+A delete instructs the set to remove a previously created cast. It is a type of soft delete where the content of the message is removed but its hash is retained forever. A user who has a copy of the deleted message can prove that the message was posted at some point by the author. 
+
+The delete message must contain the hash of the cast being removed and omit all the other properties. The set can then remove the cast and its contents from the network, which is desirable from a user perspective. The set ensures that the delete message does not reference itself, and that it is only applied as a remove operation if it has a timestamp higher than that of the message it references. 
+
+```ts
+type CastDeleteBody = {
+  targetCastHash: string;
+  schema: 'farcaster.xyz/schemas/v1/cast-delete';
+};
+```
+
+### 4.2.4 Embeds
+A data structure used to embed URIs within a cast. Clients should hydrate these URIs and show an inline preview when rendering the cast in the UI. 
```

Comment:
Why? Might be worth simply adding "to create a richer experience". And "should" is also strong, maybe "can" ?